components/usage/pkg/db/cost_center.go

```diff
@@ -263,3 +263,50 @@ func (c *CostCenterManager) ListLatestCostCentersWithBillingTimeBefore(ctx conte
 
 	return results, nil
 }
+
+func (c *CostCenterManager) ResetUsage(ctx context.Context, cc CostCenter) (CostCenter, *Usage, error) {
+	if cc.BillingStrategy != CostCenter_Other {
+		return CostCenter{}, nil, fmt.Errorf("cannot reset usage for Billing Strategy %s for Cost Center ID: %s", cc.BillingStrategy, cc.ID)
+	}
+
+	entity, _ := cc.ID.Values()
+
+	var spendingLimit int32
+	switch entity {
+	case AttributionEntity_Team:
+		spendingLimit = c.cfg.ForTeams
+	case AttributionEntity_User:
+		spendingLimit = c.cfg.ForUsers
+	default:
+		return CostCenter{}, nil, fmt.Errorf("cannot reset usage for unknown attribution entity ID: %s", cc.ID)
+	}
+
+	now := time.Now().UTC()
+
+	// Default to 1 month from now, if there's no nextBillingTime set on the record.
+	nextBillingTime := now.AddDate(0, 1, 0)
+	if cc.NextBillingTime.IsSet() {
+		nextBillingTime = cc.NextBillingTime.Time().AddDate(0, 1, 0)
+	}
+
+	// Create a synthetic Invoice Usage record, to reset usage
+	usage, err := c.ComputeInvoiceUsageRecord(ctx, cc.ID)
+	if err != nil {
+		return CostCenter{}, nil, fmt.Errorf("failed to compute invocie usage record for AttributonID: %s: %w", cc.ID, err)
+	}
+
+	// All fields on the new cost center remain the same, except for CreationTime and NextBillingTime
+	newCostCenter := CostCenter{
+		ID:              cc.ID,
+		SpendingLimit:   spendingLimit,
```

Comment:
why do you touch the spending limit here? shouldn't we just set `cc.SpendingLimit`?

Reply:
I did consider just carrying it over, but at the moment, there isn't a use-case for keeping the SpendingLimit the same as before (such that it would be different to the defaults). This ensures that it does reset it to the defaults (and sets any updated defaults). We've already forgotten to configure the spending limit defaults so existing cost centers had 1B. This would've ensured that it does reset after we do update the default spending limit.

We can change it, but we'll need to manually go and fix the 1B for all individual users, and whenever we change the default spending limit (and hence free credit allowance), we'll also need to do a bulk update.

Reply:
Ah, that's why you do that. Sounds ok to me. Could you add a comment for the reader?

Reply:
Done.